lib/cylc/mp_pool.py

```diff
@@ -102,6 +102,8 @@ def _run_command(
                 if ret_code is not None:
                     cmd_result['OUT'], cmd_result['ERR'] = proc.communicate()
                     cmd_result['EXIT'] = ret_code
+            # Wait on the job (and avoid zombie processes).
+            proc.communicate()
```

Comment:
This seems to fix the problem, but:
 1. Why was it only occurring for failed jobs?  
 2. Also, the block staring at line 100 above is? meant to handle the case that the job PID is not identified from the first 10 lines of stdout, but I don't understand how it does that.... it will do nothing if the job hasn't finished already as  ```proc.poll()``` doesn't block.